### ISSUE TITLE
Filter global channels based on type

### ISSUE THREAD

**Issue by Some-Yes-Man** [RocketChat/feature-requests#374]
**Is your feature request related to a problem? Please describe.**
As our RocketChat is taking off more and more channels and, more importantly, discussion are created.
For new users (employees) it is virtually impossible to get an overview of all exisiting public channels because the discussions in public channels clutter the global channel list.

**Describe the solution you'd like**
In the 'Directory' for 'Channels' I would like to see toggle buttons to filter by channel type (just like the admin page has, only more beautiful ^^)

**Describe alternatives you've considered**
I actually tried to change our local installation of RocketChat, but couldn't get it working, most likely because the JS files are minimzed and hard to change :P I tried to filter down the list of channels by type (the info is in there^^).

**Additional context**
The issue I'm talking about. Discussions all over the place. Not helpful for users trying to find public channels worth looking at.
![foo](https://user-images.githubusercontent.com/56762082/89556971-6b558880-d812-11ea-85a2-a11d4515693f.png)

**Comment by larkinson** [comment]
Yes, this is critical to allow channels to be properly organized if Rocket.Chat team wants their product to be competitive against other solutions on the market.

**Comment by TheApo** [comment]
+1

That is a really critical bug/feature for organisations!
